clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -27,8 +32,22 @@ private object ApiClient {
               uri.getQuery,
               uri.getFragment
              )
-    else
+    } else if (storageType == StorageTypeAzure) {
+
+      /** regex to catch url of type: https://StorageAccountName.blob.core.windows.net/Container[/BlobName] and convert it to abfs url */
+      val regex = "https?://([^.]+)(\\.[^/]+)(?:/([^/]+)(.+))?"
```

Comment:
Another way to do it, and one that I would prefer: Parse the URL.  Now you have a `host` part (that you can split on the first dot to get groups 1 and 2), followed by a path (that you can split on the first slash to get groups 3 and 4).  It should be more readable.

Context: JWZ said:
> Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.

(Who's JWZ?  JWZ wrote Netscape Navigator (which became Netscape Communicator, which became Firefox).  Before that he wrote XEmacs (which died so that Emacs could continue to live).  Go to https://www.jwz.org/hacks/ and search for `gdb-highlight.el`.)